pkg/asset/tls/ironictls.go

```diff
@@ -0,0 +1,64 @@
+package tls
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"net"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+// IronicTLSCert is the asset that generates the key/cert pair that is used for
+// enabling TLS for virtual media in ironic.
+type IronicTLSCert struct {
+	SelfSignedCertKey
+}
+
+var _ asset.WritableAsset = (*IronicTLSCert)(nil)
+
+// Dependencies returns the dependency of the the cert/key pair, which includes
+// the parent CA, and install config if it depends on the install config for
+// DNS names, etc.
+func (a *IronicTLSCert) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the cert/key pair based on its dependencies.
+func (a *IronicTLSCert) Generate(ctx context.Context, dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+	hostname := internalAPIAddress(installConfig.Config)
+
+	cfg := &CertCfg{
+		Subject:      pkix.Name{CommonName: "ironic", OrganizationalUnit: []string{"openshift"}},
+		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		Validity:     ValidityOneDay,
+	}
+
+	if installConfig.Config.Platform.BareMetal != nil {
+		vips := installConfig.Config.BareMetal.APIVIPs
+		if installConfig.Config.BareMetal.BootstrapProvisioningIP != "" {
+			vips = append(vips, installConfig.Config.BareMetal.BootstrapProvisioningIP)
+		}
+		cfg.IPAddresses = []net.IP{}
+		cfg.DNSNames = []string{hostname}
+		for _, vip := range vips {
+			cfg.IPAddresses = append(cfg.IPAddresses, net.ParseIP(vip))
+			cfg.DNSNames = append(cfg.DNSNames, vip)
```

Comment:
I don't think the VIPs should appear in the DNSNames, only in the IPAddresses.

Reply:
This is copied from MCS, not sure why they did it, but probably for a reason. Anyway, nothing is going to do host validation in reality.

Reply:
I was wondering about the DNSNames as well; will drop it out.